great_ape_safe/ape_api/cow.py

```diff
@@ -95,6 +101,46 @@ def _sell(
             buy_amount_after_fee = mantissa_buy
         else:
             buy_amount_after_fee = int(int(fee_and_quote["quote"]["buyAmount"]) * coef)
+            has_cg_price = True
+            buy_symbol, buy_decimals = buy_token.symbol(), buy_token.decimals()
+            sell_symbol, sell_decimals = sell_token.symbol(), sell_token.decimals()
+
+            try:
+                buy_token_id = self.cg_coin_list[buy_symbol.lower()]
+                sell_token_id = self.cg_coin_list[sell_symbol.lower()]
+            except KeyError:
+                has_cg_price = False
+                cow_sell_rate = (buy_amount_after_fee / 10 ** buy_decimals) / (
+                    mantissa_sell / 1e18
+                )
+                if not Confirm.ask(
+                    f"No cg rate found. Continue with cow rate of {cow_sell_rate} {sell_symbol}/{buy_symbol}?"
+                ):
+                    raise
+
+            if has_cg_price:
+                prices = self.cg.get_price([buy_token_id, sell_token_id], "usd")
+                buy_token_price = prices[buy_token_id]["usd"]
+                sell_token_price = prices[sell_token_id]["usd"]
+
+                cow_sell_rate = (buy_amount_after_fee / 10 ** buy_decimals) / (
+                    mantissa_sell / 1e18
+                )
+                cg_sell_rate = (
+                    (((mantissa_sell / 1e18) * sell_token_price) / buy_token_price)
+                    * coef
+                    / (mantissa_sell / 1e18)
+                )
+                pct_diff = (cow_sell_rate - cg_sell_rate) / cg_sell_rate
+
+                if abs(pct_diff) > self.pct_diff_threshold:
+                    print(f"cow rate: {cow_sell_rate} {sell_symbol}/{buy_symbol}")
```

Comment:
here is not the rate expressed instead on: `{buy_symbol}/{sell_symbol}`?

Reply:
yup, fixed in 5aa4523

Reply:
i dont think so. `cg_sell_rate` is calced as sell/buy on L129. eg:
selling 300 fxs
buying 2692 dai
rate is ~.11
ref https://explorer.cow.fi/orders/0x2ce41c34009c6b9b978764546a660bf165e2061db042cca8e2e2d48aa90dafebd0a7a8b98957b9cd3cfb9c0425abe44551158e9e63eb2898

Reply:
ya realized this too after further testing, reverted in a61edf2

Reply:
petrovska is right here, since:
```
cow_sell_rate = (buy_amount / buy_divisor) / (
    mantissa_sell / sell_divisor
)
```
eg
```
cow rate (before slippage): 9.972969031236483 FXS/DAI
coingecko rate: 10.032728902261416 FXS/DAI
cow rate is lower than cg by 0.6%, continue? [y/n]: 
```

which should be DAI/FXS (ie 10 dai per fxs)

Reply:
we are just comparing rates, not sure how important which asset is being bought/sold is on the ordering here. 
i think the confusion comes from how we are interpreting "ASSET1/ASSET2". coingecko uses this interpretation for example:
[ETH/USDC -> 1550](https://www.coingecko.com/en/coins/ethereum/usd)
[USDC/ETH -> 0.00064](https://www.coingecko.com/en/coins/usd-coin/eth)
basically saying ASSET1 is equal to this many ASSET2

the other way to interpret is like you said as "this many ASSET1 is equal to ASSET 2". this way seems more intuitive if you are reading it more literally from left to right, but the other interpretation is what ive seen exchanges use 🤷

regardless on interpretation, I think what we are also trying to avoid is having it spit out something like 0.0006 ETH/USDC as opposed to 1550 ETH/USDC (this depends on if we are selling or buying the stablecoin currently). so maybe the ordering should be decided programmatically: 
if price of ASSET1 > ASSET2 then have the rate of ASSET1 denominated in ASSET2 else other way around

Reply:
i see the rationale, somehow got mix-up on how the rate was calculate and how it will be more intuitive to express it, but like keeping it up like cg interprets it

would not say that its req to have a programmatically way to decide how the ratio is presented, depends on the order itself sometimes may be looking like 0.xxxx, others 1.xxx etc depending on the pricing of asset sold vs bought